src/components/date-range/date-range.test.tsx

```diff
@@ -0,0 +1,1125 @@
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      endDateProps={{ value: "01/01/2001" }}
+    />
+  );
+
+  expect(screen.getByRole("textbox", { name: "start" })).toHaveValue(
+    "10/10/2016"
+  );
+  expect(screen.getByRole("textbox", { name: "end" })).toHaveValue(
+    "01/01/2001"
+  );
+});
+
+test("should support passing custom class names via `startDateProps` and `endDateProps`", () => {
+  render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      startDateProps={{ className: "foo" }}
+      endDateProps={{ className: "bar" }}
+    />
+  );
+
+  expect(screen.getByRole("textbox", { name: "start" })).toHaveClass("foo");
+  expect(screen.getByRole("textbox", { name: "end" })).toHaveClass("bar");
+});
+
+test("should support rendering the component with `data-` tags", () => {
+  render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      data-element="foo"
+      data-role="bar"
+    />
+  );
+  const dateRange = screen.getByTestId("bar");
+
+  expect(dateRange).toHaveAttribute("data-component", "date-range");
+  expect(dateRange).toHaveAttribute("data-element", "foo");
+});
+
+test("should set the `data-element` attribute internally on each individual date input", () => {
+  render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      startDateProps={{ "data-role": "start" }}
+      endDateProps={{ "data-role": "end" }}
+    />
+  );
+
+  expect(screen.getByTestId("start")).toHaveAttribute(
+    "data-element",
+    "start-date"
+  );
+  expect(screen.getByTestId("end")).toHaveAttribute("data-element", "end-date");
+});
+
+test("should support ref as an object on the 'start' input", () => {
+  const ref = { current: null };
+  render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      startRef={ref}
+    />
+  );
+
+  expect(ref.current).toBe(screen.getByRole("textbox", { name: "start" }));
+});
+
+test("should support ref as a callback on the 'start' input", () => {
+  const ref = jest.fn();
+  render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      startRef={ref}
+    />
+  );
+
+  expect(ref).toHaveBeenCalledWith(
+    screen.getByRole("textbox", { name: "start" })
+  );
+});
+
+test("should set the ref on the 'start' input to null after the component is unmounted", () => {
+  const ref = { current: null };
+  const { unmount } = render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      startRef={ref}
+    />
+  );
+  unmount();
+
+  expect(ref.current).toBeNull();
+});
```

Comment:
**question**: are these tests asserting a ref is cleared required for coverage? If not, I'm wondering if asserting this is necessary given its a core React feature?

Reply:
I'm not sure tbh, they are just part of the set of test @robinzigmond added when he worked on adding support to our components so I've just taken the approach of replicating them whenever I've refactored the tests

Reply:
it's a fair point. I don't even remember adding these tbh 😅 . While I think the value of similar tests for refs in other components is maybe questionable, I think here it looks like there's something worth testing as `startRef` is one of our own props, not something built-in to React, so it seems fair to test that the ref ends up being added to the right element. When in doubt I'd rather have too many tests than too few.

But definitely not a disaster if this is removed and isn't needed for coverage, so I won't insist on it.